packages/blockchain-link/src/workers/apiBase.ts

```diff
@@ -0,0 +1,261 @@
+import WebSocket from 'ws';
+import { createDeferred, Deferred } from '@trezor/utils/lib/createDeferred';
+import { TypedEmitter } from '@trezor/utils/lib/typedEventEmitter';
+
+import { CustomError } from '@trezor/blockchain-link-types/lib/constants/errors';
+
+interface Subscription<T> {
+    id: string;
+    type: T;
+    callback: (result: any) => void;
+}
+
+interface Options {
+    url: string;
+    timeout?: number;
+    pingTimeout?: number;
+    keepAlive?: boolean;
+    agent?: WebSocket.ClientOptions['agent'];
+    headers?: WebSocket.ClientOptions['headers'];
+}
+
+const DEFAULT_TIMEOUT = 20 * 1000;
+const DEFAULT_PING_TIMEOUT = 50 * 1000;
+
+type EventMap = { [event: string]: any };
+
+type WsEvents = {
+    error: string;
+    disconnected: undefined;
+};
+
+export abstract class APIBase<T extends EventMap> extends TypedEmitter<T & WsEvents> {
```

Comment:
there is already `base.ts` (BaseWorker) file in workers dir. maybe lets name it `BaseWebsocket` ?

Reply:
Both of you are right, fixed & squashed.